packages/cli/src/services/activeWorkflows.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Service } from 'typedi';
+
+import type { User } from '@db/entities/User';
+import { SharedWorkflowRepository } from '@db/repositories/sharedWorkflow.repository';
+import { WorkflowRepository } from '@db/repositories/workflow.repository';
+import { ActivationErrorsService } from '@/ActivationErrors.service';
+import { BadRequestError } from '@/errors/response-errors/bad-request.error';
+import { Logger } from '@/Logger';
+
+@Service()
+export class ActiveWorkflowsService {
+	constructor(
+		private readonly logger: Logger,
+		private readonly workflowRepository: WorkflowRepository,
+		private readonly sharedWorkflowRepository: SharedWorkflowRepository,
+		private readonly activationErrorsService: ActivationErrorsService,
+	) {}
+
+	async getAllActiveIds() {
+		const activationErrors = await this.activationErrorsService.getAll();
+		const activeWorkflowIds = await this.workflowRepository.getActiveIds();
+		return activeWorkflowIds.filter((workflowId) => !activationErrors[workflowId]);
+	}
+
+	async getAllActiveIdsForUser(user: User) {
```

Comment:
Nit - Already clear from the arg name and type.

```suggestion
	async getAllActiveIdsFor(user: User) {
```